test/testsuites/csctargetnamespacetests.go

```diff
@@ -4,122 +4,152 @@ import (
+)
 
-// testCscWithNonExistingTargetNamespace creates context and calls the
-// cscWithNonExistingTargetNamespace test case.
-func testCscWithNonExistingTargetNamespace(t *testing.T) {
+// CscTargetNamespace is a test suit that confirms that targetNamespace values are handled appropriately.
+func CscTargetNamespace(t *testing.T) {
 	ctx := test.NewTestCtx(t)
 	defer ctx.Cleanup()
 
 	// Get global framework variables.
 	f := test.Global
 
-	// Run tests.
-	if err := cscWithNonExistingTargetNamespace(t, f, ctx); err != nil {
-		t.Fatal(err)
-	}
-}
-
-// cscWithNonExistingTargetNamespace is a test case that creates a CatalogSourceConfig
-// with a targetNamespace that doesn't exist and ensures that the status is updated to reflect the
-// nonexistant namespace. The test then creates the targeted namespace and ensures that the
-// CatalogSourceConfig is properly reconciled.
-func cscWithNonExistingTargetNamespace(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
 	// Get test namespace.
-	namespace, err := test.NewTestCtx(t).GetNamespace()
+	namespace, err := ctx.GetNamespace()
 	if err != nil {
-		return fmt.Errorf("Could not get namespace: %v", err)
+		t.Fatalf("Could not get namespace: %v", err)
 	}
 
-	// nonExistingTargetNamespaceCscName is the name of the CatalogSourceConfig that points
-	// to a non-existing targetNamespace.
-	nonExistingTargetNamespaceCscName := "non-existing-namespace-operators"
-
-	// targetNamespace is the non-existing target namespace.
-	targetNamespace := "non-existing-namespace"
-
 	// Create a new CatalogSourceConfig with a non-existing targetNamespace.
-	nonExistingTargetNamespaceCsc := &operator.CatalogSourceConfig{
+	nonExistingTargetNamespaceCsc := &marketplace.CatalogSourceConfig{
 		TypeMeta: metav1.TypeMeta{
-			Kind: operator.CatalogSourceConfigKind,
+			Kind: marketplace.OperatorSourceKind,
 		}, ObjectMeta: metav1.ObjectMeta{
-			Name:      nonExistingTargetNamespaceCscName,
+			Name:      cscName,
 			Namespace: namespace,
 		},
-		Spec: operator.CatalogSourceConfigSpec{
+		Spec: marketplace.CatalogSourceConfigSpec{
 			TargetNamespace: targetNamespace,
-			Packages:        "descheduler",
+			Packages:        "camel-k-marketplace-e2e-tests",
 		}}
-	err = helpers.CreateRuntimeObject(f, ctx, nonExistingTargetNamespaceCsc)
+
+	// CatalogSourceConfig created with nil ctx, must be deleted manually before test suite exits.
+	err = helpers.CreateRuntimeObject(f, nil, nonExistingTargetNamespaceCsc)
 	if err != nil {
-		return err
+		t.Fatalf("Unable to create test CatalogSourceConfig: %v", err)
 	}
 
 	// Check that we created the CatalogSourceConfig with a non-existing targetNamespace.
-	resultCatalogSourceConfig := &operator.CatalogSourceConfig{}
-	err = helpers.WaitForResult(t, resultCatalogSourceConfig, namespace, nonExistingTargetNamespaceCscName)
+	resultCatalogSourceConfig := &marketplace.CatalogSourceConfig{}
+	err = helpers.WaitForResult(t, resultCatalogSourceConfig, namespace, cscName)
```

Comment:
Aren't you doing this check in the `helpers.WaitForExpectedPhaseAndMessage` which is called below in the `t.Run` functions?

Reply:
My reasoning is that the `t.Run` functions below are supposed to be ran with the expectation that the `CSC` exists. If the CSC isn't created the user should be looking in this method. If we remove this check, the output from the e2e tests will tell people to look in the other test cases instead.

Reply:
That's a great point

Reply:
Can we document this thought process in comments?

Reply:
Will do @kevinrizza